next/components/search/search-multicheckbox-facet.tsx

```diff
@@ -21,10 +21,10 @@ export function MultiCheckboxFacet({
 }: FacetViewProps) {
   const { t } = useTranslation();
   return (
-    <fieldset className={clsx(className, "mb-4")}>
-      <legend className="mb-5 text-heading-xs font-bold text-steelgray">
+    <section className={clsx(className, "mb-4")}>
+      <div className="mb-5 text-heading-xs font-bold text-steelgray">
         {label}
-      </legend>
+      </div>
```

Comment:
This `label` is actually pretty important - it's telling the user what the group of checkboxes is actually for, so I think we should make that clearer in some way. Is there a reason the fieldset+legend approach didn't work well? That's what MDN recommends for this kind of thing: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/fieldset#try_it

If we want it to be clear that the search results are filtered by these values, we could also add to the aria-label a bit. Something like `aria-label={t("filter-by-label", { label })`, implemented in common.json like `"filter-by-label": "Filter by {{label}}"`